### PR TITLE
improved analysis options files and fix strong mode issues

### PR DESCRIPTION
**Pull request by devoncarew**
- fix some strong mode issues in the dart code
- update the analysis options files to use strong mode

### CODE REVIEW

File: analysis_options.yaml [hunk 1]
```diff
@@ -1,7 +1,8 @@
 analyzer:
```
Comment (mit-mit) [inline analysis_options.yaml]: Why generally is this file different from https://github.com/flutter/flutter/blob/master/analysis_options.yaml#L24 ?

File: tool/extract.dart [hunk 2]
```diff
@@ -112,5 +111,5 @@ void clean() {
   Iterable<FileSystemEntity> files = exampleDir
       .listSync()
       .where((FileSystemEntity entity) => entity is File && entity.path.endsWith('.dart'));
-  files.forEach((File file) => file.deleteSync());
+  files.forEach((FileSystemEntity file) => file.deleteSync());
```
Comment (mit-mit) [inline tool/extract.dart]: Why is this needed given the where clause above filters to files only?

Comment (devoncarew) [reply tool/extract.dart]: This was to fix a strong mode issue, but I updated it to make the definition of `files` more specific (`Iterable<File>`).

File: analysis_options.yaml (hunk 1 above)
Comment (devoncarew) [reply analysis_options.yaml]: I think just historically, and to avoid copying and pasting copies of the flutter analysis options. I updated this to try using an analysis options `include:` directive; we'll see if travis is happy with that.

Comment (Hixie) [reply analysis_options.yaml]: wow, i didn't know include: was a thing. If it is we may be able to simplify a number of other files.

Comment (devoncarew) [reply analysis_options.yaml]: Yup, developed I think for use by Flutter, but not yet rolled out in use. We'll give it a go here and see how it works. /cc @danrubel
